Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -511,6 +606,49 @@ private extension PurchasesOrchestrator {
         deviceCache.setOfferingsCacheTimestampToNow()
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func purchase(sk2Package: Package) async -> Result<(CustomerInfo, Bool), Error> {
+        guard let sk2ProductDetails = sk2Package.productDetails as? SK2ProductDetails else {
+            // todo: use custom error
```

Comment:
👍

Reply:
https://app.shortcut.com/revenuecat/story/11158/set-up-custom-error-for-when-unwrapping-sk2productdetails-fails